### PR TITLE
Move directory directives under a special key.

### LINKED ISSUE #1078: Move directory directives under a special key
Looking into #1062 i realize that it is very hard to validate our configuration because we  deliberately made it very ambiguous if something is a directory directive or a smell detector.
Validating our configuration AND handling most edgecases will be a nightmare.
I think we can make it easier:
- introduce a new top level key called "directories" analoguous to [exclude_paths](https://github.com/troessner/reek#configuration-options) under which we put the directives
- the change should be trivial for users but would make our lifes easier. 
- users can still fuck up the config but since we can make solid assumptions (e.g. everything under "directories" is a directory directive) we can show the users far better and helpful error messages that will help him figure it out on his own. E.g. ""foo/bar/baz is not a valid smell detector. Are you trying to set up a directory directive? Please see...."
- this would be a breaking change of course

Wdyt?

### PR DESCRIPTION
**Pull request by troessner**
Fixes #1078

### CODE REVIEW

**Comment by troessner** [comment]
Ready for review!

File: spec/reek/configuration/app_configuration_spec.rb [hunk 1]
```diff
@@ -99,11 +102,10 @@
...
       let(:configuration_as_hash) do
-        {
-          directory => { TooManyStatements: { max_statements: 8 } },
-          :IrresponsibleModule => { enabled: false },
-          :TooManyStatements => { max_statements: 15 }
-        }
+        { described_class::DIRECTORIES_KEY =>
+            { directory => { TooManyStatements: { max_statements: 8 } } },
+          'IrresponsibleModule' => { enabled: false },
+          'TooManyStatements' => { max_statements: 15 } }
```
Comment (mvz) [inline spec/reek/configuration/app_configuration_spec.rb]: I'm not sure why RuboCop isn't complaining but the formatting here is a little off.

File: spec/reek/configuration/app_configuration_spec.rb [hunk 2]
```diff
@@ -99,11 +102,10 @@
...
 
       let(:configuration_as_hash) do
-        {
-          directory => { TooManyStatements: { max_statements: 8 } },
-          :IrresponsibleModule => { enabled: false },
-          :TooManyStatements => { max_statements: 15 }
-        }
+        { described_class::DIRECTORIES_KEY =>
+            { directory => { TooManyStatements: { max_statements: 8 } } },
+          'IrresponsibleModule' => { enabled: false },
```
Comment (mvz) [inline spec/reek/configuration/app_configuration_spec.rb]: Is there a particular reason for changing the key from symbol to string here?

(I notice that we may have an inconsistency here, in that symbols work for the smell types, but perhaps not for the other keys).

Comment (troessner) [reply spec/reek/configuration/app_configuration_spec.rb]: >> Is there a particular reason for changing the key from symbol to string here?

No, good point, changed it back.

File: spec/reek/configuration/app_configuration_spec.rb (hunk 1 above)
Comment (troessner) [reply spec/reek/configuration/app_configuration_spec.rb]: Moved around the brackets a bit - better?

Comment (mvz) [reply spec/reek/configuration/app_configuration_spec.rb]: Yes, better!

**Comment by troessner** [comment]
@mvz I also added two comments to the README so we do not confuse Reek 4 users that come back to this documentation to look stuff up:
* [comment nr. 1](https://github.com/troessner/reek/pull/1318/files#diff-04c6e90faac2675aa89e2176d2eec7d8R317)
* [comment nr. 2](https://github.com/troessner/reek/pull/1318/files#diff-04c6e90faac2675aa89e2176d2eec7d8R347)

We can then remove those comments in a year or so. Normally this would be the point to introduce version dependent documentation (apart from the official release notes) but I'd like to stay away from this as long as possible since this will complicate the documentation quite a bit. Since this seems to be the only breaking change that might confuse returning Reek 4 users I think we're good.

**Pull request merged**